src/dotty/tools/dotc/typer/TypeAssigner.scala

```diff
@@ -470,7 +470,7 @@ trait TypeAssigner {
     tree.withType(sym.nonMemberTermRef)
 
   def assignType(tree: untpd.Annotated, annot: Tree, arg: Tree)(implicit ctx: Context) =
-    tree.withType(AnnotatedType(arg.tpe, Annotation(annot)))
+    tree.withType(AnnotatedType(arg.tpe.widen, Annotation(annot)))
```

Comment:
Actually there is one issue created for this:

https://github.com/lampepfl/dotty/issues/1330

A lot of places depend on widening, currently there is no simple fix.